### ISSUE TITLE
IGeometry, (IPuntal, ILineal, IPolygonal Improvements)

### ISSUE THREAD

**Issue by juliusfriedman** [NetTopologySuite/NetTopologySuite#457]
```cs
public interface IGeometry, 
//IEnumerable<IGeometry> Could also be a separate marker I gues...
{
 bool IsEmpty {get;}
 int NumGeometries {get;}
//Perhaps GeometryType etc
//IEnumerable<Geometry> Geometries {get;} //Or enumerator...
}
```

Then please add ```IGeometry``` as the base implementation of ```IPuntal, ILineal, IPolygonal```.

Alternatively add properties to each of the aforementioned interfaces.

The scenario is when you have a few different ```ILineal``` objects, some of which may not even be ```Geometry``` but would otherwise link a ```Geometry``` or provide a facade so need at least some of the basic methods and properties on ```Geometry``` without having to be a ```Geometry```.

As an added benefit this would also allow overriding the methods which are not marked virtual if needed by a derived implementation.

**Comment by airbreather** [comment]
`ILineal` and friends in JTS are [marker interfaces](https://www.geeksforgeeks.org/marker-interface-java/), presumably to be able to tell what dimension a `Geometry` is without specifically caring about whether it's a single or multi.  For example, if it's `ILineal`, then it's either a `LineString` or a `MultiLineString`.  These were ported to NTS as-is.

Can you please explain further what goal you are trying to accomplish by using these interfaces with non-`Geometry` objects?

**Comment by juliusfriedman** [comment]
Its easier to explain in terms of Geometry, e.g. to add ComoundCurve I would use a GeometryCollectionEx adding support for count of contained types along with other utility methods / properties such as Lineality (amount of ILineals), from there I would derive CompoundCurve and CircularLineString which is probably more useful than the non geometry case.

For non Geometry types I might have a Manifold which is not a Geometrical object that can be measured discretely such a Brain, I have a inclination of what space is but not where I am in space, I can say what exists there and how to approximate the Location  or Surface curvature but not exactly how large it is, (Infinite)... 

In other words what I think would be a Geometry underneath Geometry with the Unknown SRID.

**Comment by juliusfriedman** [comment]
While your at it can you also please see about making Geometry IEnumerable so one doesn't have to write the trival GeometryEnumerator and then use what should already be there IMHO :)

**Comment by bjornharrtell** [comment]
Seems stale to me, let me know if still of interest.
